libs/feature-workspaces/src/lib/workspaces/workspaces.component.ts

```diff
@@ -22,13 +28,38 @@ import { WorkspacesService } from '../workspaces.service';
     ])
   ]
 })
-export class WorkspacesComponent {
+export class WorkspacesComponent implements OnInit {
   constructor(
     readonly settings: Settings,
-    readonly workspacesService: WorkspacesService
+    readonly workspacesService: WorkspacesService,
+    private readonly contextualActionBarService: ContextualActionBarService,
+    private readonly matDialog: MatDialog
   ) {}
 
   trackByPath(_: number, w: WorkspaceDescription) {
     return w.path;
   }
+
+  ngOnInit() {
+    this.contextualActionBarService.breadcrumbs$.next([
+      { title: 'Select an Angular Workspace' }
+    ]);
+  }
+
+  importExistingWorkspace() {
+    this.workspacesService.selectExistingWorkspace().subscribe(result => {
```

Comment:
Do we need a `pipe(first())` to make sure the stream completes... or manual unsubscribe on destroy?

Reply:
Added pipe to selectExistingWorkspace() itself